packages/ens-controller/src/EnsController.ts

```diff
@@ -86,6 +113,46 @@ export class EnsController extends BaseControllerV2<
         ...state,
       },
     });
+
+    if (provider && getCurrentChainId) {
+      const chainId = getCurrentChainId();
+      const network = (CHAIN_ID_TO_NETWORK_ID_MAP as any)[
```

Comment:
We should avoid casting if possible, and we _especially_ should avoid using `any` (which completely disables type checking).

Reply:
Agreed. I tried to help Salah with this earlier today and couldn't quite make sense of this on the call. I'll see if I can offer a better suggestion here.